Source/Plugins/OrientationAnalysis/Test/GridifyTest.cpp

```diff
@@ -0,0 +1,225 @@
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE
+ * USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ *
+ *
+ * ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~ */
+
+#pragma once
+
+#include <QtCore/QCoreApplication>
+#include <QtCore/QFile>
+#include <unordered_map>
+
+#include "SIMPLib/Common/SIMPLibSetGetMacros.h"
+#include "SIMPLib/DataArrays/DataArray.hpp"
+#include "SIMPLib/Filtering/FilterFactory.hpp"
+#include "SIMPLib/Filtering/FilterManager.h"
+#include "SIMPLib/Filtering/FilterPipeline.h"
+#include "SIMPLib/Plugin/ISIMPLibPlugin.h"
+#include "SIMPLib/Plugin/SIMPLibPluginLoader.h"
+#include "SIMPLib/SIMPLib.h"
+
+#include "SIMPLib/Filtering/QMetaObjectUtilities.h"
+#include "SIMPLib/Geometry/ImageGeom.h"
+#include "SIMPLib/Geometry/TriangleGeom.h"
+
+#include "UnitTestSupport.hpp"
+//#include "OrientationAnalysisTestFileLocations.h"
+
+class GridifyTest
+{
+  const QString m_filterName{"Gridify"};
+  const QString m_amName{"Grid"};
+  const QString m_aaName{"Layout"};
+  const DataContainerArray::Pointer m_dca;
+  AbstractFilter::Pointer m_filter{nullptr};
+
+  std::unordered_map<std::string, std::vector<uint16_t>> checkValues{
+      {"I",   {0, 0, 0}},
+      {"II",  {2, 0, 0}},
+      {"III", {2, 3, 0}},
+      {"IV",  {0, 2, 0}},
+      {"V",   {3, 1, 0}},
+      {"VI",  {1, 1, 1}},
+      {"VII", {3, 3, 2}},
+  };
+
+  int ValidateData()
+  {
+    int wrongData{0};
+    for(const auto& eachDC : m_dca->getDataContainers())
+    {
+      if (!eachDC->doesAttributeMatrixExist(m_amName))
+      {
+        continue;
+      }
+      uint16_t xTestComp{eachDC->getAttributeMatrix(m_amName)
+            ->getAttributeArrayAs<UInt16ArrayType>(m_aaName)
+            ->getValue(0)};
+      uint16_t yTestComp{eachDC->getAttributeMatrix(m_amName)
+            ->getAttributeArrayAs<UInt16ArrayType>(m_aaName)
+            ->getValue(1)};
+      uint16_t zTestComp{eachDC->getAttributeMatrix(m_amName)
+            ->getAttributeArrayAs<UInt16ArrayType>(m_aaName)
+            ->getValue(2)};
+      uint16_t xCheckComp{checkValues[eachDC->getName().toStdString()][0]};
+      uint16_t yCheckComp{checkValues[eachDC->getName().toStdString()][1]};
+      uint16_t zCheckComp{checkValues[eachDC->getName().toStdString()][2]};
+      if (!((xCheckComp == xTestComp) && (yCheckComp == yTestComp) && (zCheckComp == zTestComp)))
+      {
+        wrongData++;
+      }
+    }
+    return wrongData;
+  }
+
+  int RemoveTestFiles()
+  {
+#if REMOVE_TEST_FILES
+    //    QFile::remove(UnitTest::GridifyTest::TestFile1);
+    //    QFile::remove(UnitTest::GridifyTest::TestFile2);
+#endif
+    return 0;
+  }
+
+  int SetUp()
+  {
+    return 0;
+  }
```

Comment:
This looks like something that either was forgotten to be implemented or does not need to be.  Consider removing it if it's unnecessary.